### ISSUE TITLE
Difficulty Replicating Performance in Monocular Depth Estimation

### ISSUE THREAD

**Issue by mathjihun** [needylove/OrdinalEntropy#7]
Hello, I have attempted to replicate the performance of the NeWCRFs model several times as per the given methodology. However, the results I obtained, with RMSE 0.334, Abs_rel 0.093, and D1 0.922, show a significant deviation from the performance presented in the paper. Moreover, these results are obtained when squaring the L2 norm of the tightness loss term, and interestingly, using the L2 norm as suggested in the paper leads to a slight decrease in performance. (the baseline performance of NeWCRFs is RMSE 0.334, Abs_rel 0.095, and D1 0.923.)

I have also applied the same approach to other models, such as PixelFormer, VPD, and others, and consistently observed performance improvements. Concerned that I might have made a mistake, I have a few questions.

1. When using OrdinalEntropy for training, is it correct to experiment with the final output of the decoder, specifically using features from e0 in the image below? 
![image](https://github.com/needylove/OrdinalEntropy/assets/87381172/cc3d3238-40db-4d6c-adf7-6750850e6cb3)
2. Is it correct to conduct experiments with a batch size of 8?
3. Could you provide information about the GPU environment during the experiments?

I appreciate your consistently valuable insights.

**Comment by needylove** [comment]
Hi, 
1. Yes, using e0
2. I just checked the code and found the batch size is set to 4. I guess I want to conduct experiments with a single GPU and thus set the batch size to 4 previously, and I just checked and realized the batch size exploited in NewCRF is 8.
3. Sorry, I can hardly find the original GPU environment information, as our server has been updated two times over the past year. I just remember the GPU is 48G memory. The performance difference may be due to my wrongly set batch size. Sorry for the inconvenience.

**Comment by mathjihun** [comment]
Thank you, your response has been very helpful. I will try with a batch size of 4.

**Comment by mathjihun** [comment]
The experiment with a batch size of 4 resulted in even lower performance. Did you use the default values for epoch and learning rate from the config setting on NeWCRFs' GitHub, which are 50 and 2e-5, respectively?

**Comment by needylove** [comment]
Hi, the learning rate is set to 1e-4, which is the same as the NeWCRF. The epoch is set to 50. I remember we followed the default setting of NeWCRF (while I forgot I changed the batch size to train models on a single gpu)

**Comment by mathjihun** [comment]
Originally, it seems that the evaluation was based on 20 epochs, and it appears that NeWCRFs' GitHub has not been updated to reflect this. All my doubts have been resolved. Thank you for your consistently excellent responses. It has been very helpful.
